Dependencies/UrlLib/Source/Apple/UrlRequest.mm

```diff
@@ -56,6 +58,11 @@ void ResponseType(UrlResponseType value)
 
             NSURLSession* session{[NSURLSession sharedSession]};
             NSURLRequest* request{[NSURLRequest requestWithURL:url]};
+            if (url == nil)
+            {
+                // Complete the task, but retain the default status code of 0 to indicate a client side error.
+                return arcana::task_from_result<std::exception_ptr>();
```

Comment:
We still need to work on getting the status text propagated back out for this to be fully correct.